docs/table/modify_table.rst

```diff
@@ -164,14 +164,33 @@ the table and replaces existing ones::
    foo     1.0    10.0
    bar     2.0    10.0
 
-:meth:`~astropy.table.Table.update` also takes care of silently :ref:`merging_metadata`::
+When using ``|=``, the other object does not need to be a |Table|, it can be
+anything that can be used for :ref:`construct_table` with a compatible number
+of rows.
 
-  >>> t1.meta
+It is also possible to use ``|`` to return the new |Table| rather than update
+in-place. ``|`` requires the other object to be a |Table|::
```

Comment:
```suggestion
It is also possible to use the ``|`` operator to merge multiple |Table| instances
into a new table::
```
I would also suggest writing an example of attempting to merge a `Table` with something that's not a `Table`. If a user triggers this error and tries to find documentation about it then an explicit example with the correct error message might make it easier to find with a search engine.